src/js/ripple/remote.js

```diff
@@ -148,155 +97,133 @@ function Remote(opts, trace) {
   }
-
   if (typeof this.local_signing !== 'boolean') {
-    throw new TypeError('Remote "local_signing" configuration is not a Boolean');
+    throw new TypeError('local_signing must be a boolean');
   }
-
   if (typeof this.local_fee !== 'boolean') {
-    throw new TypeError('Remote "local_fee" configuration is not a Boolean');
+    throw new TypeError('local_fee must be a boolean');
   }
-
   if (typeof this.local_sequence !== 'boolean') {
-    throw new TypeError('Remote "local_sequence" configuration is not a Boolean');
+    throw new TypeError('local_sequence must be a boolean');
   }
-
-  if (!/^(undefined|number)$/.test(typeof opts.ping)) {
-    throw new TypeError('Remote "ping" configuration is not a Number');
+  if (typeof this.canonical_signing !== 'boolean') {
+    throw new TypeError('canonical_signing must be a boolean');
   }
-
-  if (!/^(undefined|object)$/.test(typeof opts.storage)) {
-    throw new TypeError('Remote "storage" configuration is not an Object');
+  if (typeof this.submission_timeout !== 'number') {
+    throw new TypeError('submission_timeout must be a number');
   }
-
-  // Fallback for previous API
-  if (!opts.hasOwnProperty('servers') && opts.hasOwnProperty('websocket_ip')) {
-    opts.servers = [
-      {
-        host:     opts.websocket_ip,
-        port:     opts.websocket_port,
-        secure:   opts.websocket_ssl,
-        trusted:  opts.trusted
-      }
-    ];
+  if (typeof this.last_ledger_offset !== 'number') {
+    throw new TypeError('last_ledger_offset must be a number');
+  }
+  if (!Array.isArray(opts.servers)) {
+    throw new TypeError('servers must be an array');
   }
 
-  (opts.servers || []).forEach(function(server) {
-    self.addServer(server);
-  });
-
-  // This is used to remove Node EventEmitter warnings
-  var maxListeners = opts.maxListeners || opts.max_listeners || 0;
+  this.setMaxListeners(this.max_listeners);
 
-  this._servers.concat(this).forEach(function(emitter) {
-    if (emitter instanceof EventEmitter) {
-      emitter.setMaxListeners(maxListeners);
-    }
+  this.servers.forEach(function(server) {
+    var connection = self.addServer(server);
+    connection.setMaxListeners(self.max_listeners);
   });
 
-  function listenerAdded(type, listener) {
-    if (type === 'transaction_all') {
-      if (!self._transaction_subs && self.isConnected()) {
-        self.request_subscribe('transactions').request();
-      }
-      self._transaction_subs += 1;
-    }
-  };
-
-  this.on('newListener', listenerAdded);
-
-  function listenerRemoved(type, listener) {
-    if (type === 'transaction_all') {
-      self._transaction_subs -= 1;
-      if (!self._transaction_subs && self.isConnected()) {
-        self.request_unsubscribe('transactions').request();
+  function listenersModified(action, event) {
+    // Automatically subscribe and unsubscribe to orderbook
+    // on the basis of existing event listeners
+    if (~Remote.TRANSACTION_EVENTS.indexOf(event)) {
```

Comment:
You're an assembly coder aren't you? :smile:

This isn't readable. I didn't even expect that the binary representation of -1 is all ones because all numbers are treated as floats in javascript.

I'd suggest _.includes